src/eth/NonceService.js

```diff
@@ -3,26 +3,27 @@ import { promisify } from '../utils';
 
 export default class NonceService extends PublicService {
   constructor(name = 'nonce') {
-    super(name, ['web3']);
+    super(name, ['web3', 'accounts']);
+    this._counts = {};
   }
 
   async connect() {
-    await this.setNextNonce();
+    await this.setCounts();
```

Comment:
what you've done here makes sense. additionally, though, we need to handle the case where the user calls `addAccount` after the Maker instance is already init'ed/connected/auth'ed.

this is for e.g. when you are in the governance dashboard and have already started the Maker instance to get read-only data, but then you add a hardware wallet.

so i think you could just initialize each wallet "just-in-time" during getNonce if you see that `this.counts[address]` is undefined.

Reply:
Good call. I'll add the "just in time" implementation before merging.